plugins/moveToArchive.koplugin/main.lua

```diff
@@ -0,0 +1,163 @@
+                        return
+                    end
+                    if self.ui.document then
+                        self.ui:onClose()
+                    end
+                    if FileManager.instance then
+                        FileManager.instance:reinit(archive_dir_path)
+                    else
+                        FileManager:showFiles(archive_dir_path)
+                    end
+                end,
+            },
+            {
+                text = _("Go to last copied/moved from folder"),
+                callback = function()
+                    if not last_copied_from_dir then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No previous folder found.")
+                         })
+                        return
+                    end
+                    if self.ui.document then
+                        self.ui:onClose()
+                    end
+                    if FileManager.instance then
+                        FileManager.instance:reinit(last_copied_from_dir)
+                    else
+                        FileManager:showFiles(last_copied_from_dir)
+                    end
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved.\nDo you want to open it from the archive folder?")
+    local doc_old_path = self:commonProcess(true, move_done_text)
+    ReadHistory:removeItemByPath(doc_old_path)
+end
+
+function MoveToArchive:copyToArchive()
+    local copy_done_text =_("Book copied.\nDo you want to open it from the archive folder?")
+    self:commonProcess(false, copy_done_text)
+end
+
+function MoveToArchive:commonProcess(is_move_process, moved_done_text)
+    if not self.ui.document then
+        UIManager:show(InfoMessage:new{
+            text = _("Only works in reading book mode.")
+         })
+        return
+    end
+    if not archive_dir_path then
+        self:showNoArchiveConfirmBox()
+        return
+    end
+    local document_full_path = G_reader_settings:readSetting("lastfile")
+    self.ui:onClose()
+    local filename
+    last_copied_from_dir, filename = util.splitFilePathName(document_full_path)
+
+    logger.dbg("MoveToArchive: last_moved/copied_from_dir :", last_copied_from_dir)
+    move_to_archive_settings:saveSetting(last_copied_from_config_key, ("%s/"):format(last_copied_from_dir))
+
+    if is_move_process then
+        FileManager:moveFile(document_full_path, archive_dir_path)
+        FileManager:moveFile(DocSettings:getSidecarDir(document_full_path), archive_dir_path)
+    else
+        FileManager:copyFileFromTo(document_full_path, archive_dir_path)
+        FileManager:copyFileFromTo(DocSettings:getSidecarDir(document_full_path), archive_dir_path)
```

Comment:
Does that work? copyFileFromTo uses /bin/cp, without any -r option to copy the directory and its content.
I still think it's just not "experimental", it's a bad idea copying the book and the docsettings, and have them in 2 places, not knowing which docsetting/progress is the active one (and the other one will be stalled - if the user makes highlights while continuing reading, they won't be in the other book docsettings).

Reply:
Copy is essentially "copy" ;) of "move" command. But with replaced "mv" to "cp". So as it works for move should work same way for copy

Reply:
"should work" ? Have you tested it?
```
/tmp/test $ touch this_is_a_file
/tmp/test $ mkdir this_is_a_dir
/tmp/test $ mkdir other_dir
/tmp/test $ mv this_is_a_file other_dir/
/tmp/test $ mv this_is_a_dir other_dir/
/tmp/test $

/tmp/test $ touch this_is_a_file2
/tmp/test $ mkdir this_is_a_dir2
/tmp/test $ mv this_is_a_file2 other_dir
/tmp/test $ cp this_is_a_dir2 other_dir/
cp: -r not specified; omitting directory 'this_is_a_dir2'
/tmp/test $ ls
other_dir  this_is_a_dir2
/tmp/test $
```
`this_is_a_dir2` would be your sidecar directory.